src/code/z_view.c

```diff
@@ -1,62 +1,603 @@
+                      view->viewport.bottomY);
+        gSPViewport(pLocal++, viewport);
+    }
+    if (arg1 & 8) {
+        projection = Graph_Alloc(gfxCtx, sizeof(Mtx));
+        LogUtils_CheckNullPointer("projection", projection, "../z_view.c", 921);
+        view->unk_E0 = projection;
+        func_801045A4(projection, -(f32)(s32)gScreenWidth * 0.5f, (f32)(s32)gScreenWidth * 0.5f,
+                      -(f32)(s32)gScreenHeight * 0.5f, (f32)(s32)gScreenHeight * 0.5f, view->fogDistance, view->zDepth,
+                      view->unk_24);
+
+        view->unk_60 = *projection;
+
+        gSPMatrix(pLocal++, projection, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_PROJECTION);
+    } else if (arg1 & 6) {
+        projection2 = Graph_Alloc(gfxCtx, sizeof(Mtx));
+        LogUtils_CheckNullPointer("projection", projection2, "../z_view.c", 932);
+        view->unk_E0 = projection2;
+        guPerspective(projection2, &view->unk_11C, view->fieldOfView,
+                      (f32)(view->viewport.rightX - view->viewport.leftX) /
+                          (f32)(view->viewport.bottomY - view->viewport.topY),
+                      view->fogDistance, view->zDepth, view->unk_24);
+
+        view->unk_60 = *projection2;
+
+        gSPPerspNormalize(pLocal++, view->unk_11C);
+        gSPMatrix(pLocal++, projection2, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_PROJECTION);
+    }
+    if (arg1 & 1) {
+        viewing = Graph_Alloc(gfxCtx, sizeof(MtxF));
+        LogUtils_CheckNullPointer("viewing", viewing, "../z_view.c", 948);
+        view->unk_E4 = viewing;
+        func_800ABE74(view->eye.x, view->eye.y, view->eye.z);
+        func_80101E34(viewing, view->eye.x, view->eye.y, view->eye.z, view->unk_34.x, view->unk_34.y, view->unk_34.z,
+                      view->unk_40.x, view->unk_40.y, view->unk_40.z);
+
+        view->unk_A0 = *viewing;
+
+        gSPMatrix(pLocal++, viewing, G_MTX_NOPUSH | G_MTX_MUL | G_MTX_PROJECTION);
+    }
+    view->unk_120 = 0;
+    *p = pLocal;
+    return 1;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800AB9EC.s")
+#endif
+
+s32 func_800ABE74(f32 eyeX, f32 eyeY, f32 eyeZ) {
+    s32 errorCode = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800ABE74.s")
+    if ((eyeX * eyeX) + (eyeY * eyeY) + (eyeZ * eyeZ) > 1073676288.0f) {
+        errorCode = 3;
+    } else {
+        f32 absEyeX;
+        f32 absEyeY;
+        f32 absEyeZ;
+        absEyeX = ABS(eyeX);
+        absEyeY = ABS(eyeY);
+        absEyeZ = ABS(eyeZ);
+        if (((18900.0f < absEyeX) || (18900.0f < absEyeY)) || (18900.0f < absEyeZ)) {
+            errorCode = 2;
+        } else if (((16000.0f < absEyeX) || (16000.0f < absEyeY)) || (16000.0f < absEyeZ)) {
+            errorCode = 1;
+        }
+    }
+    if (errorCode != 0) {
+        osSyncPrintf("\x1b[31m");
+        // Is too large
+        osSyncPrintf("eye が大きすぎます eye=[%8.3f %8.3f %8.3f] error=%d\n", eyeX, eyeY, eyeZ, errorCode);
+        osSyncPrintf("\x1b[m");
```

Comment:
VT macros should be used instead of \x1b strings, which don't really make sense on their own.